src/pygeom.c

```diff
@@ -168,6 +167,32 @@ char get_geom(GeometryObject *obj, GEOSGeometry **out) {
     }
 }
 
+/* Check if the input is a GeometryObject or Py_None. Returns 0 on error, 1 on success. */
+char check_geom(GeometryObject *obj) {
+    if (!PyObject_IsInstance((PyObject *) obj, (PyObject *) &GeometryType)) {
```

Comment:
If we would simplify this check to an exact check

```
if (Py_TYPE(obj) != (PyTypeObject *) GeometryType)
```

We might be able to do this with the GIL released?